src/payments-methods/express/payment-methods-express.js

```diff
@@ -14,19 +14,48 @@ import { renderAmazonButton } from '../../renderAmazonButton';
  * @returns React component
  */
 const AmazonPayExpressBtn = ( props ) => {
+	const estimatedOrderAmount = calculateEstimatedOrderAmount( props );
+
 	useEffect( () => {
-		renderAmazonButton( '#pay_with_amazon_express', 'express', null );
+		renderAmazonButton( '#pay_with_amazon_express', 'express', null, estimatedOrderAmount );
 	}, [] );
 
 	return <div id="pay_with_amazon_express" />;
 };
 
+/**
+ * Returns the estimated order amount button attribute.
+ * @param {object} props
+ * @returns {object}|null Estimated order amount button attribute.
+ */
+const calculateEstimatedOrderAmount = ( props ) => {
+	const { billing } = props;
+	const { currency } = billing;
+
+	const stringCartTotal     = String( billing.cartTotal.value );
+	const cartTotalLength     = stringCartTotal.length;
+	const decimals            = currency.minorUnit;
+	const checkOutValue       = stringCartTotal.slice( 0, cartTotalLength - decimals ) + '.' + stringCartTotal.slice( cartTotalLength - decimals );
```

Comment:
Not sure what's happening here. Can you elaborate, please?

Reply:
@kalessil i added doc comments in every step of the function. Could you let me know if the functionality is clear now ?